internal/provider/resource_access_key.go

```diff
@@ -0,0 +1,214 @@
+	return &schema.Resource{
+		Description: `Provides an Infra access key. This resource can be used to create and manage connector access keys.
+
+-> This resource requires Infra server version 0.20.0 or higher.`,
+
+		CreateContext: resourceAccessKeyCreate,
+		ReadContext:   resourceAccessKeyRead,
+		DeleteContext: resourceAccessKeyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "The access key's unique identifier.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"name": {
+				Description:      "The access key's name. If omitted, a name will be automatically generated. Identity provider names may include letters (uppercase and lowercase), numbers, underscores `_`, hyphens `-`, and periods `.`.",
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateStringIsName(),
+			},
+			"secret": {
+				Description: "The access key secret.",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"expires_in": {
+				Description:      `The total amount of time before the access key expires. Format is a duration string, a sequence of decimal numbers, each with optional fraction and a unit suffix, such as "300s" or "2h45m". Valid time units are "s", "m", "h". Default is 8766h0m0s.`,
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateStringIsDuration(),
+				DiffSuppressFunc: DurationDiffSuppressFunc(),
+				ConflictsWith: []string{
+					"expires_at",
+				},
+			},
+			"expires_at": {
+				Description:      `The date-time when the access key will expire. Format is a RFC3339 timestamp, e.g. "2006-01-02T15:04:05Z07:00."`,
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsRFC3339Time),
+				ConflictsWith: []string{
+					"expires_in",
+				},
+			},
+			"inactivity_timeout": {
+				Description:      `The amount of time before the access key expires if left unused. If the access key is used before it expires, it will be renewed for the same duration. Format is a duration string, a sequence of decimal numbers, each with optional fraction and a unit suffix, such as "300s" or "2h45m". Valid time units are "s", "m", "h". If value is greater than or equal to the remaining lifetime of the access key, the access key will not be renewed. Default is 72h0m0s.`,
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateStringIsDuration(),
+				DiffSuppressFunc: DurationDiffSuppressFunc(),
+			},
+		},
+	}
+}
+
+func resourceAccessKeyCreate(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
```

Comment:
Is there a better alternative here than to accept `any` (along with a bunch of type conversion below)? Or is this standard for terraform resource methods